snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py

```diff
@@ -62,14 +62,174 @@ def __init__(self, replica_metadata: dict, uuid: Optional[str] = None):
                 f"SNOWSHU_{self.uuid}_{self.replica_meta['name'].upper()}"
             )
 
+    def set_replica_prefix(self, replica_prefix: str):
+        SnowflakeAdapter.replica_prefix = replica_prefix
+
+    def build_catalog(self, thread_workers: int = 4, **kwargs) -> Set[Relation]:  # pylint: disable=arguments-differ
+        """
+        Builds and returns a set of Relations present in Snowflake replicas
+        from databases that start with the given prefix.
+
+        Args:
+            thread_workers (int): Number of threads to use for concurrent fetching.
+
+        Returns:
+            Set[Relation]: A set of Relation objects from databases matching the prefix.
+        """
+
+        del kwargs  # Surpress unused variable warning
+
+        catalog = set()
+
+        def accumulate_relations(database: str):
+            if not database.startswith(self.replica_prefix):
+                logger.debug(
+                    f"Skipping database '{database}' as it does not start with prefix '{self.replica_prefix}'."
+                )
+                return
+            try:
+                schemas = self._get_all_schemas(database)
+                for schema in schemas:
+                    try:
+                        relations = self._get_relations_from_database(database, schema)
+                        catalog.update(relations)
+                    except sqlalchemy.exc.SQLAlchemyError as exc:
+                        logger.error(
+                            f"Error fetching relations from schema '{schema}' in database '{database}': {exc}"
+                        )
+            except sqlalchemy.exc.SQLAlchemyError as exc:
```

Comment:
I would suggest moving `except` block after `schemas = self._get_all_schemas(database)` to reduce nesting

Reply:
agree